### PR TITLE
🚀 koa2 experimental support

### PR DESCRIPTION
**Pull request by nicolasgramlich**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 dist: trusty
 language: node_js
 node_js:
-  - '6'
-  - '5'
-  - '4'
+  - '7'
```
Comment (SylvainEstevez) [inline .travis.yml]: I'd like restypie to remain compatible until v4. We're not the only ones using it.

File: src/route.js [hunk 2]
```diff
@@ -93,12 +93,12 @@ module.exports = class Route {
 
       case Restypie.RouterTypes.KOA_ROUTER:
         /* istanbul ignore next */
-        return function * restypieRouteHandler (next) {
-          if (this.request.body) this.req.body = this.request.body;
-          this.req.params = this.params; // Copy params so that we don't have to parse them
-          this.req.query = this.query;
-          this.state.bundle = new Restypie.Bundle({ req: this.req, res: this.res, url });
-          yield next;
+        return async function restypieRouteHandler (ctx, next) {
```
Comment (SylvainEstevez) [inline src/route.js]: You won't be able to use this syntax when downgrading to node@4

File: .travis.yml (hunk 1 above)
Comment (nicolasgramlich) [inline .travis.yml]: Hm... `@next` :D ?

File: src/route.js (hunk 2 above)
Comment (nicolasgramlich) [inline src/route.js]: Is there any other way?

Comment (jonahglover) [inline src/route.js]: This change is for `koa-router` 7.x which requires`koa` 2.x which requires a  minimum of node v7.6.0 so I don't think you'd be using this release with node@4 anyway...

**Comment by jakemh** [comment]
why node 4? https://github.com/nodejs/LTS#lts-schedule1

node 4 maintenance ends `2018-04-01`

File: src/route.js (hunk 2 above)
Comment (jakemh) [inline src/route.js]: https://github.com/nodejs/LTS#lts-schedule1

node 4 will reach end of life 2018-04-01. also we should be using a luxe fork anyway.

Comment (jakemh) [inline src/route.js]: also koa 2 doesn't require 7.6, just recommends it. it just requires async/await support

**Comment by SylvainEstevez** [comment]
So 2 solutions :
- https://github.com/SylvainEstevez/Restypie/blob/a364d97c7b27a4f0f1d073a1918693bca6924eba/README.md ~1day of work
- Have a Luxe fork

I'll let that up to you @nicolasgramlich

**Pull request closed without merging**